lib/logstash/filters/drop.rb

```diff
@@ -26,6 +39,6 @@ def register
 
   public
   def filter(event)
-    event.cancel
+    event.cancel if rand < (@sample / 100.0)
```

Comment:
to prevent the extra computation of `rand` at every drop, maybe we can 
check for whether sampling is even enabled with something like this and 
short circuit the computation if unnecessary

```ruby
event.cancel if (@sample and rand < (@sample / 100.0))
```